Vagrantfile

```diff
@@ -2,16 +2,20 @@
 # vi: set ft=ruby :
 
 LINUX_BASE_BOX = "bento/ubuntu-16.04"
+LINUX_DOCKER_BOX = "tknerr/baseimage-ubuntu-16.04"
```

Comment:
why can't we use the official 16.04 image instead?

Reply:
Because the official ubuntu image - `ubuntu/xenial64` supports only `virtualbox` provider. And `bento/ubuntu-16.04` doesn't support `docker` provider. That's why I've added the new vagrant image with `docker` provider support.

Reply:
Instead of adding a random docker vm, let's just install docker on the linux base box?

```
  config.vm.define "dockervm" do |d|
    d.vm.box = "bento/ubuntu-18.04"
    d.vm.provider :virtualbox

    d.vm.provision "shell", inline:<<-SHELL
    sudo apt-get update
    sudo apt-get install \
        apt-transport-https \
        ca-certificates \
        curl \
        software-properties-common
    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo apt-key add -
    sudo apt-key fingerprint 0EBFCD88
    sudo add-apt-repository \
       "deb [arch=amd64] https://download.docker.com/linux/ubuntu \
       $(lsb_release -cs) \
       stable"
    sudo apt-get update
    sudo apt-get install docker-ce -y
    SHELL
  end
```

Reply:
Oops, you're wanting docker as the provider, not a box with docker installed. I see.